docs/website/docs/reference/tuning.md

```diff
@@ -0,0 +1,126 @@
+---
+icon: octicons/meter-16
+---
+
+# Tuning
+
+This page documents support for IREE dispatch tuning. The compiler supports
+both default and user-provided tuning specs that override compiler heuristics
+that guide dispatch code generation.
```

Comment:
High level feedback on this intro:

* Some motivating background information or benchmark metrics would help put this into context. Why would a user want to run tuning? For example: "on this hardware, applying tuning improves performance of these matrix multiply operations by XX% and performance of a full ABC model by YY%". Is tuning always useful? What does it entail? How long does it take? When is it worth using compared to other optimization methods?
* The top of the page looks a bit like a wall of text. Diagrams, charts, or icons would make this more approachable

![image](https://github.com/user-attachments/assets/a576398e-aedd-4f9e-a782-f47cbaaeccf8)

Fine to merge and then iterate though.

Reply:
Added motivation and split the description of specs into a new subsection